micro_sam/training/simple_sam_trainer.py

```diff
@@ -0,0 +1,51 @@
+import random
+
+from . import SamTrainer
+
+
+class SimpleSamTrainer(SamTrainer):
+    """Trainer class for creating a simple SAM trainer for limited prompt-based segmentation.
+    """
+    def __init__(
+        self,
+        use_points: bool = False,
+        use_box: bool = False,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
```

Comment:
Please recheck what we had in the MedSAM trainer, where the relevant super kwargs `n_sub_iter=1, mask_prob=0` where set already.

Reply:
Ahha yes, makes sense to set these defaults. I'll do this for the new `MedSAMTrainer` as suggested below, by inheriting `SimpleSamTrainer` and setting the defaults for MedSAM.